bee-network/bee-autopeering/src/config.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Autopeering configuration.
+//!
+//! ## JSON Example
+//! ```json
+//! "autopeering": {
+//!     "enabled": true,
+//!     "bindAddress": "0.0.0.0:14626",
+//!     "entryNodes": [
+//!          "/dns/lucamoser.ch/udp/14826/autopeering/4H6WV54tB29u8xCcEaMGQMn37LFvM1ynNpp27TTXaqNM",
+//!          "/dns/entry-hornet-0.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaPHdAn7eueBnXtikZMwhfPXaeGJGXDt4RBuLuGgb",
+//!          "/dns/entry-hornet-1.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaJJqMd5CQvv1A61coSQCYW9PNT1QKPs7xh2Qg5K2",
+//!          "/dns/entry-mainnet.tanglebay.com/udp/14626/autopeering/iot4By1FD4pFLrGJ6AAe7YEeSu9RbW9xnPUmxMdQenC"
+//!     ],
+//!     "entryNodesPreferIPv6": true,
+//! }
+//! ```
+//!
+//! ## TOML Example
+//!
+//! ```toml
+//! [autopeering]
+//! enabled = true
+//! bind_address = "0.0.0.0:14626"
+//! entry_nodes = [
+//!     "/dns/lucamoser.ch/udp/14826/autopeering/4H6WV54tB29u8xCcEaMGQMn37LFvM1ynNpp27TTXaqNM",
+//!     "/dns/entry-hornet-0.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaPHdAn7eueBnXtikZMwhfPXaeGJGXDt4RBuLuGgb",
+//!     "/dns/entry-hornet-1.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaJJqMd5CQvv1A61coSQCYW9PNT1QKPs7xh2Qg5K2",
+//!     "/dns/entry-mainnet.tanglebay.com/udp/14626/autopeering/iot4By1FD4pFLrGJ6AAe7YEeSu9RbW9xnPUmxMdQenC"
+//! ]
+//! entry_nodes_prefer_ipv6 = true
+//! ```
+
+use crate::multiaddr::AutopeeringMultiaddr;
+
+use serde::{Deserialize, Serialize};
+
+use std::{
+    fmt::Debug,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+};
+
+// TODO: watch out for possible constification regarding `SocketAddr::new()`.
+const AUTOPEERING_BIND_ADDR_DEFAULT: IpAddr = IpAddr::V4(Ipv4Addr::UNSPECIFIED);
```

Comment:
Shouldn't localhost be the default ?

Reply:
No, the default should be "anything you find", and not any particular IP address. You can redefine localhost to basically any IP you want. (UNSPECIFIED is 0.0.0.0). It's done exactly the same in Hornet/Hive.